modules/biobambam/bammerge/main.nf

```diff
@@ -0,0 +1,41 @@
+process BIOBAMBAM_BAMMERGE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::biobambam=2.0.183" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/biobambam:2.0.183--h9f5acd7_1':
+        'quay.io/biocontainers/biobambam:2.0.183--h9f5acd7_1' }"
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("${prefix}.bam")       ,emit: bam
+    tuple val(meta), path("${indexfilename}")    ,optional:true, emit: bam_index
+    tuple val(meta), path("${md5filename}")      ,optional:true, emit: checksum
+
+    path "versions.yml"                       ,emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    indexfilename = args.contains("indexfilename=") ? (args =~ /indexfilename=([^\s]+)/)[0][1] : ""
```

Comment:
I think this capture would be better understood if it's just based on a file extension. This is over-complicating the file just to capture the name. 
Since processes are isolated, it would be quite safe to use "*.bai" as the output line.

Reply:
Sure, I can do just the `bai`, but I wanted to leave the option open for more esoteric filenames, since there are no restrictions there.
Same for the md5sum file.

What do you suggest? Just assume default extensions (`bai` and `md5`)?

Reply:
Personally I think it would be good to enforce conventional filename name extensions.

Reply:
No problem, I'll fix it up